db/migrate/20190523093215_add_topic_id_to_user_exports.rb

```diff
@@ -0,0 +1,5 @@
+class AddTopicIdToUserExports < ActiveRecord::Migration[5.2]
+  def change
+    add_column :user_exports, :topic_id, :integer
```

Comment:
Does this need to be `null: false`? It seems like the links for user export are only exposed via a PM so I would think that we need to enforce that `UserExport` are always tied to a topic.

Reply:
I was worried about existing rows in `user_exports` table. I will do something like `change_column_null :table_name, :column_name, false.

Reply:
Actually `null: false` might not be possible because the topic is created _after_ the UserExport record is created and then we update the `topic_id` column, same as what we do for `upload_id`.

Reply:
Hmm that is tricky. Is there a reason we send a system message for the link instead of an email like we do for admins to download backups?

Reply:
No specific reason. I guess this was how we envisioned this feature to work in initial spec i.e. downloading archive without needing to leave Discourse UX.

Reply:
Note that we also send system message (with backup logs) when taking a backup and link Admin to admin/backups page to download that file. This is done for additional security purpose I believe.

Reply:
I guess we just need to create the topic and upload before creating the `UserExport`?

Reply:
That's not trivial to do as upload contains file name which includes `UserExport` id and topic has file download link which can't be generated without creating user_export record and upload first.

Reply:
Ah icic. I guess we can just leave it as nullable then.